libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/internal/MapboxNavigationViewApi.kt

```diff
@@ -0,0 +1,157 @@
+package com.mapbox.navigation.dropin.internal
+
+import com.mapbox.bindgen.Expected
+import com.mapbox.bindgen.ExpectedFactory
+import com.mapbox.geojson.Point
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.base.route.NavigationRoute
+import com.mapbox.navigation.dropin.NavigationViewApi
+import com.mapbox.navigation.ui.app.internal.Store
+import com.mapbox.navigation.ui.app.internal.destination.Destination
+import com.mapbox.navigation.ui.app.internal.destination.DestinationAction
+import com.mapbox.navigation.ui.app.internal.endNavigation
+import com.mapbox.navigation.ui.app.internal.extension.dispatch
+import com.mapbox.navigation.ui.app.internal.navigation.NavigationState
+import com.mapbox.navigation.ui.app.internal.navigation.NavigationStateAction
+import com.mapbox.navigation.ui.app.internal.routefetch.RoutePreviewAction
+import com.mapbox.navigation.ui.app.internal.routefetch.RoutePreviewState
+import com.mapbox.navigation.ui.app.internal.routefetch.RoutesAction
+import com.mapbox.navigation.ui.app.internal.tripsession.TripSessionStarterAction
+
+@ExperimentalPreviewMapboxNavigationAPI
+internal class MapboxNavigationViewApi(
+    private val store: Store
+) : NavigationViewApi() {
+
+    override fun startFreeDrive() {
+        store.dispatch(endNavigation())
+    }
+
+    override fun startDestinationPreview(point: Point) {
+        store.dispatch(DestinationAction.SetDestination(Destination(point)))
+        store.dispatch(NavigationStateAction.Update(NavigationState.DestinationPreview))
+    }
+
+    override fun startRoutePreview(): Expected<Throwable, Unit> =
+        runCatchingExpected {
+            checkDestination()
+            checkPreviewRoutes()
+
+            store.dispatch(NavigationStateAction.Update(NavigationState.RoutePreview))
+        }
+
+    override fun startRoutePreview(routes: List<NavigationRoute>): Expected<Throwable, Unit> =
+        runCatchingExpected {
+            require(routes.isNotEmpty()) { "routes cannot be empty" }
+            val point = checkNotNull(routes.first().getDestination()) {
+                "destination not found in a given route"
+            }
+
+            setDestination(point)
+            setPreviewRoutes(routes)
+            startRoutePreview()
```

Comment:
🤔 If this and similarly others below can be extracted into `ThunkActions`? As far as I understand, that's why we introduced it to handle a group of actions.
@tomaszrybakiewicz

Reply:
@tomaszrybakiewicz when you get a chance can you pitch in here?

Reply:
Of course, we can introduce `ThunkActions` for these groups of actions. Technically, every method in this class could be turned into a ThunkAction. However, I haven't found other use cases where those actions would also be dispatched, therefore I left that refactoring till later.